### DISCUSSION TITLE
Programmatically quit or abort in a dynamic form?

### DISCUSSION THREAD

**Discussion by puyt** [charmbracelet/huh#430] [Q&A]
I'm trying to load the select options dynamically, as demonstrated in one of the examples using `OptionsFunc`. Since the fetch may fail, I want to exit gracefully and print an appropriate error message.

```
OptionsFunc(func() []huh.Option[string] {
	items, err := fetchItems()
	if err != nil {
		fmt.Println("Failed to fetch items")
		os.Exit(0)
	}

	return items
}, nil)
```

**Comment by Alphasite** [comment]
Similar question, how do you gracefully handle failures inside dynamic forms?
